src/fmu/dataio/export/rms/inplace_volumes.py

```diff
@@ -200,6 +200,47 @@ def _convert_table_from_legacy_to_standard_format(
         table = self._add_missing_columns_to_table(table)
         return self._set_table_column_order(table)
 
+    def _is_column_missing_in_table(self, column: str) -> bool:
+        """Check if a column is present in the final dataframe and has values"""
+        return column not in self._dataframe or self._dataframe[column].isna().all()
+
+    def _validate_table(self) -> None:
+        """
+        Validate that the final table with volumes is according to the standard
+        defined for the inplace_volumes product.
+        """
+        _logger.debug("Validating the dataframe...")
+
+        has_oil = "oil" in self._dataframe[_FLUID_COLUMN].values
+        has_gas = "gas" in self._dataframe[_FLUID_COLUMN].values
+
+        # check that one of oil and gas fluids are present
+        if not (has_oil or has_gas):
+            raise RuntimeError(
+                "One or both 'oil' and 'gas' needs to be selected as 'Main types'"
+                "in the volumetric job. Please update and rerun the volumetric job "
+                "before export."
+            )
+
+        # create list of missing or non-defined required columns
+        missing_calculations = []
+        for col in ["BULK", "PORV", "HCPV"]:
+            if self._is_column_missing_in_table(col):
+                missing_calculations.append(col)
+
+        if has_oil and self._is_column_missing_in_table("STOIIP"):
+            missing_calculations.append("STOIIP")
+
+        if has_gas and self._is_column_missing_in_table("GIIP"):
+            missing_calculations.append("GIIP")
```

Comment:
Would be nice to convert these strings to enums https://github.com/equinor/fmu-dataio/issues/920